tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/ServerInterceptor.java

```diff
@@ -24,13 +27,15 @@
  */
 public abstract class ServerInterceptor implements HandlerInterceptor {
 
+	private static final String REQUEST_ID = "requestId";
 	private volatile String origThreadName;
 
 	@Override
 	public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
 			throws Exception {
 		origThreadName = Thread.currentThread().getName();
 		Thread.currentThread().setName(getThreadName());
+		MDC.put(REQUEST_ID, UUID.randomUUID().toString());
```

Comment:
I've also used MDC for this in other dev projects. It's probably still the best option. 

From experience it can get quite expensive to generate a lot of UUIDs especially if the transactions are very small. Can we have the UUID as a string at startup and then an AtomicLong that we increment and append?

Reply:
Good point! Should be changed. Introduced a process UUID which is postponed with an `AtomicLong` in a format which should be parsable by some tooling if need be.

Let me know you thoughts.